pkg/node/grpc_server.go

```diff
@@ -1,35 +1,43 @@
-package signer
+package node
 
 import (
 	"context"
 	"fmt"
 	"time"
 
+	"github.com/strangelove-ventures/horcrux/pkg/pcosigner"
+	"github.com/strangelove-ventures/horcrux/pkg/types"
+
 	"github.com/hashicorp/raft"
-	"github.com/strangelove-ventures/horcrux/signer/proto"
+	"github.com/strangelove-ventures/horcrux/pkg/proto"
 )
 
-var _ proto.CosignerGRPCServer = &GRPCServer{}
+// Enures that GRPCServer implements the proto.CosignerGRPCServer interface.
+var _ proto.ICosignerGRPCServerServer = &GRPCServer{}
 
 type GRPCServer struct {
-	cosigner           *LocalCosigner
-	thresholdValidator *ThresholdValidator
-	raftStore          *RaftStore
-	proto.UnimplementedCosignerGRPCServer
+	cosigner           *pcosigner.LocalCosigner // The "node's" LocalCosigner
+	thresholdValidator *ThresholdValidator      // The "node's" ThresholdValidator
+	raftStore          *RaftStore               // The "node's" RaftStore
+	// Promoted Fields is embedded to have forward compatiblitity
+	proto.UnimplementedICosignerGRPCServerServer
 }
 
+// NewGRPCServer returns a new GRPCServer.
 func NewGRPCServer(
-	cosigner *LocalCosigner,
+	cosigner *pcosigner.LocalCosigner,
 	thresholdValidator *ThresholdValidator,
 	raftStore *RaftStore,
 ) *GRPCServer {
 	return &GRPCServer{
+		// TODO: This is a hack to get around the fact that the cosigner is not a?
```

Comment:
Is this TODO still relevant?

Reply:
Was mainly a placeholder for me for a "future" fix.